adam/flask_backend/app.py

```diff
@@ -93,7 +99,9 @@ def get_scene() -> Any:
         / EXPERIMENTS_TESTING_DIR_NAME
         / testing_curriculum
         / SITUATION_DIR_NAME.format(num=scene_number)
-    )
+    ).resolve()
+    if not is_relative_to(experiment_dir, LEARNERS_DIR):
+        return {"message": "Directory out of range"}
```

Comment:
When the user tries to retrieve data in an out-of-range directory, should we always err out with `HTTPStatus.BAD_REQUEST`? Currently, this response is inconsistent with the response in `get_image()

Reply:
I think the different responses make sense. In this case we have more specific/restricted parameters so that we know the user is requesting info on a particular situation. With `get_image()` the user gives an arbitrary path within `DATA_DIR`, so it is harder to give a useful error message. It would make sense to return an error code as well rather than 200 OK -- [I think](https://flask.palletsprojects.com/en/1.1.x/quickstart/#about-responses) the syntax would be:

```python
return {"message": "Directory out of range"}, HTTPStatus.BAD_REQUEST
```